### PR TITLE
Add some useful scalafmt rules

### PR DESCRIPTION
**Pull request by kciesielski**
This PR extends scalafmt a little bit. I tried to select a few rules that are very common and not really controversial.

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+newlines.implicitParamListModifierForce      = [before, after]
+project.git                                  = true
+rewrite.rules                                = [PreferCurlyFors, RedundantBraces, RedundantParens, SortImports]
+spaces.inImportCurlyBraces                   = true
+style                                        = defaultWithAlign
+indentOperator.exemptScope                   = all
+rewriteTokens {
+  "⇒" = "=>"
+  "→" = "->"
+  "←" = "<-"
```
Comment (lukaszlenart) [inline .scalafmt.conf]: ❤️

**Review by rucek — APPROVED** [review]

**Comment by adamw** [comment]
I was keeping this file as simple as possible for a purpose - to avoid any formatting wars. We simply use the defaults so there's no place for discussion ;)

So if we're to add something, it would be great to have some justification for the rules (maybe the file can be commented?)

**Comment by kciesielski** [comment]
@adamw I added some comments, could you re-review?

**Comment by kciesielski** [comment]
BTW I'm OK with keeping the file super-simple without any rules, in order to avoid imposing anything. If you prefer it that way, just close this PR, no offense will be taken :)

**Comment by adamw** [comment]
@kciesielski I don't know ;) I like using the defaults as it doesn't cuts any discussions short. I guess the question is - if these changes are "good", why aren't they default?

**Comment by codesurf42** [comment]
I would love to have some defaults based on some common sense. In general, for maintenance and peace of life - I prefer rules that makes editing and merging easier with less possible conflicts, but still formatting code in a readable way. Less unicode, less vertical alignment. But everything here could be controversial :)

**Comment by adamw** [comment]
@codesurf42 from my experience the defaults are fine (except for line width ;) )

**Comment by kciesielski** [comment]
I agree that these rules are subjective, but I'll try to defend at least:
```
RedundantBraces, RedundantParens, SortImports
```
These rules remove unneeded braces and parens, which is super-rarely unwanted. And who doesn't want sorted imports? :)

**Comment by adamw** [comment]
@kciesielski agreed on those 3 :) it also brings you closer to Scala 3, and indeed sorting imports is sth that annoyed me for some time ;)

**Comment by codesurf42** [comment]
:+1: for sorted imports (saves time and money)

**Comment by kciesielski** [comment]
@adamw I reduced to mentioned three rules; please re-review.

**Comment by adamw** [comment]
Thanks :)

**Pull request merged**